packages/@lwc/babel-plugin-component/src/decorators/index.js

```diff
@@ -163,39 +140,119 @@ function generateInvalidDecoratorError(path) {
     }
 }
 
+function collectDecoratorPaths(bodyItems) {
+    return bodyItems.reduce((acc, bodyItem) => {
+        const decorators = bodyItem.get('decorators');
+        if (decorators && decorators.length) {
+            acc.push(...decorators);
+        }
+        return acc;
+    }, []);
+}
+
+function getDecoratorMetadata(decoratorPath) {
+    const expressionPath = decoratorPath.get('expression');
+
+    let name;
+    if (expressionPath.isIdentifier()) {
+        name = expressionPath.node.name;
+    } else if (expressionPath.isCallExpression()) {
+        name = expressionPath.node.callee.name;
+    } else {
+        throw generateInvalidDecoratorError(decoratorPath);
+    }
+
+    const propertyName = decoratorPath.parent.key.name;
+    const decoratedNodeType = getDecoratedNodeType(decoratorPath);
+
+    return {
+        name,
+        propertyName,
+        path: decoratorPath,
+        decoratedNodeType,
+    };
+}
+
+function getMetadataObjectPropertyList(t, decoratorMetas, classBodyItems) {
+    const list = [
+        ...api.transform(t, decoratorMetas, classBodyItems),
+        ...track.transform(t, decoratorMetas),
+        ...wire.transform(t, decoratorMetas),
+    ];
+
+    const fieldNames = classBodyItems
+        .filter((field) => field.isClassProperty({ computed: false, static: false }))
+        .filter((field) => !field.node.decorators)
```

Comment:
We only allow the 3 LWC decorators.

Reply:
Mmm, I see. As a more general topic (and somewhat unrelated to this PR), I understand that we enforce that you can only use the 3 lwc decorators in an lwc component but do not necessarily agree with it.

From the purity perspective (and just my opinion), suppose I want to add a `logger` decorator that logs whenever a method is called. To use it, I must transform before the lwc one, and it's expected to modify only the code that involves that decorator. I like to think the same way about the lwc plugin: we just modify what we know and is intended for lwc; the rest, we leave it as is, and in theory, it shouldn't matter the order.

Reply:
That's a good point. I would prefer to make that change in a different PR though because it changes some assumptions we make in several places (e.g., there can be more than one decorator on a field). Could you create an issue for it?